contracts/access/AccessController.sol

```diff
@@ -164,21 +146,22 @@ contract AccessController is IAccessController, AccessManagedUpgradeable, UUPSUp
         // This includes initiating calls to these modules and receiving calls from them.
         // Additionally, it can modify Permissions settings.
         AccessControllerStorage storage $ = _getAccessControllerStorage();
-        if (
-            to != address(this) &&
-            !IModuleRegistry($.moduleRegistry).isRegistered(to) &&
-            !IModuleRegistry($.moduleRegistry).isRegistered(signer)
-        ) {
-            revert Errors.AccessController__BothCallerAndRecipientAreNotRegisteredModule(signer, to);
-        }
         // Must be a valid IPAccount
         if (!IIPAccountRegistry($.ipAccountRegistry).isIpAccount(ipAccount)) {
             revert Errors.AccessController__IPAccountIsNotValid(ipAccount);
         }
-        // Owner can call all functions of all modules
+        // Owner can call any contracts either registered module or unregistered/external contracts
         if (IIPAccount(payable(ipAccount)).owner() == signer) {
             return;
         }
+        if (
+            to != address(this) &&
```

Comment:
"If we are calling to an address that is not the IPAccount, and such destination address is not a registered module, and the signer is not a registered module", revert.

I'm not sure what are we preventing here. I'm going to try using reversed logic, please @kingster-will correct me if I'm wrong

- If the destination we are calling is the IPAccount address -> OK
or
- destination is a registered module -> OK
or
- signer is a registered module -> OK

OK here means that we are going to check if there are granular or module level permissions in the lines below.

In which cases the signer would a registered module?

Reply:
I believe the first check is checking if "to" is Access Controller, not IP Account.

Reply:
I see.
`to != address(this)` is for checking it whether call to AccessController but IPAccount.
It verifies whether either the signer or the to is a registered module.

Reply:
please add proper comments in the code @kingster-will

Reply:
Added comments to clarify it.